source/opt/debug_info_manager.cpp

```diff
@@ -514,16 +517,16 @@ Instruction* DebugInfoManager::AddDebugValueWithIndex(
   return added_dbg_value;
 }
 
-void DebugInfoManager::AddDebugValueIfVarDeclIsVisible(
+bool DebugInfoManager::AddDebugValueIfVarDeclIsVisible(
     Instruction* scope_and_line, uint32_t variable_id, uint32_t value_id,
     Instruction* insert_pos,
     std::unordered_set<Instruction*>* invisible_decls) {
   auto dbg_decl_itr = var_id_to_dbg_decl_.find(variable_id);
-  if (dbg_decl_itr == var_id_to_dbg_decl_.end()) return;
+  if (dbg_decl_itr == var_id_to_dbg_decl_.end()) return false;
 
+  bool modified = false;
   for (auto* dbg_decl_or_val : dbg_decl_itr->second) {
-    if (scope_and_line &&
-        !IsDeclareVisibleToInstr(dbg_decl_or_val, scope_and_line)) {
+    if (!IsDeclareVisibleToInstr(dbg_decl_or_val, scope_and_line)) {
```

Comment:
What happened with checking if `scope_and_line` is not null?

Reply:
The assert in line [446](https://github.com/KhronosGroup/SPIRV-Tools/pull/4002/files/b1bc0c10a2d83a3852d0b07d803b11c8e6abb525#diff-6e5ffb311b0364a87e9794b09d9efa5d5a431fa651b137dce6b1715c8bc9be53L446) will fail. I want to deny null for `scope_and_line` for now (maybe we want to allow it in the future if I realize there are some corner cases).

I will add the same assert in this function for the readability.

Reply:
OK